Modules/Relationships/src/Relationships.Infrastructure/Persistence/Database/Repository/RelationshipTemplatesRepository.cs

```diff
@@ -47,14 +48,24 @@ public async Task Delete(Expression<Func<RelationshipTemplate, bool>> filter, Ca
         return template;
     }
 
-    public async Task<DbPaginationResult<RelationshipTemplate>> FindTemplatesWithIds(IEnumerable<RelationshipTemplateId> ids, IdentityAddress identityAddress, PaginationFilter paginationFilter,
+    public async Task<DbPaginationResult<RelationshipTemplate>> FindTemplatesWithIds(IEnumerable<RelationshipTemplateQuery> queries, IdentityAddress activeIdentity, PaginationFilter paginationFilter,
         CancellationToken cancellationToken, bool track = false)
     {
+        var queriesList = queries.ToList();
+
+        Expression<Func<RelationshipTemplate, bool>> whereClause = template => false;
+
+        foreach (var inputQuery in queriesList)
+        {
+            whereClause = whereClause
+                .Or(RelationshipTemplate.HasId(RelationshipTemplateId.Parse(inputQuery.Id))
+                    .And(RelationshipTemplate.CanBeCollectedWithPassword(activeIdentity, inputQuery.Password))
+                    .And(RelationshipTemplate.CanBeCollectedBy(activeIdentity)));
```

Comment:
I wonder: couldn't we just add the check for `CanBeCollectedBy` once for the whole query, instead of once for every item of the query?